lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -734,16 +742,63 @@ void OCLToSPIRVBase::visitCallConvert(CallInst *CI, StringRef MangledName,
   if (auto *VecTy = dyn_cast<VectorType>(SrcTy))
     SrcTy = VecTy->getElementType();
   auto IsTargetInt = isa<IntegerType>(TargetTy);
+  auto TargetSigned = DemangledName[8] != 'u';
 
   std::string TargetTyName(
       DemangledName.substr(strlen(kOCLBuiltinName::ConvertPrefix)));
   auto FirstUnderscoreLoc = TargetTyName.find('_');
   if (FirstUnderscoreLoc != std::string::npos)
     TargetTyName = TargetTyName.substr(0, FirstUnderscoreLoc);
+
+  // Validate target type name
+  std::regex Expr("([a-z]+)([0-9]*)$");
+  std::smatch DestTyMatch;
+  if (!std::regex_match(TargetTyName, DestTyMatch, Expr))
+    return;
+
+  // The first sub_match is the whole string; the next
+  // sub_match is the first parenthesized expression.
+  std::string DestTy = DestTyMatch[1].str();
+
+  // check it's valid type name
+  SmallVector<StringRef, 11> ValidTypes = {"float", "double", "half",   "char",
+                                           "uchar", "short",  "ushort", "int",
+                                           "uint",  "long",   "ulong"};
+  bool IsValidType = false;
+  for (auto I : ValidTypes) {
```

Comment:
Probably this can be done faster if we make `ValidTypes` a static set of strings (e.g., `static std::unordered_set<std::string> ValidTypes = {...}`) that initializes just once and then search in it using smth like `if (auto It = ValidTypes.find(I); It != ValidTypes.end())`.